### PR TITLE
whitelist PV usage metrics 

### PR DESCRIPTION
**Pull request by nickcurie**
## What does this PR change?
Whitelists `kubelet_volume_stats_used_bytes`

## Does this PR rely on any other PRs?

- [OC#1403](https://github.com/opencost/opencost/pull/1403)
- [KCM#963](https://github.com/kubecost/kubecost-cost-model/pull/963)

## How does this PR impact users? (This is the kind of thing that goes in release notes!)
Whitelisted PV usage metrics into Prometheus

## Links to Issues or ZD tickets this PR addresses or fixes

- 
- 

## How was this PR tested?
Locally

### CODE REVIEW

File: cost-analyzer/charts/prometheus/values.yaml
Comment (michaelmdresser) [inline cost-analyzer/charts/prometheus/values.yaml:1264]: IIRC `kubelet_volume_stats_used_bytes` is in alpha status as a metric. Based on the [documentation](https://kubernetes.io/docs/concepts/cluster-administration/system-metrics/#metric-lifecycle), that means that the metric is subject to changes or removal at any time.

Is the code you've written using this metric resilient to the removal/change of the metric? In other words, if the metric were to disappear tomorrow, would the rest of the Asset/Disk pipeline still continue to function as it does today?

For product/eng leadership (@dwbrown2, @nikovacevic): Do we have guidelines on building Kubecost functionality on top of alpha/beta K8s functionality? For context, we've previously had contention over a beta->v1 migration/breaking change with cluster-turndown's CRD: https://github.com/kubecost/cost-analyzer-helm-chart/pull/1444#issuecomment-1145092419.

File: cost-analyzer/charts/prometheus/values.yaml
Comment (michaelmdresser) [inline cost-analyzer/charts/prometheus/values.yaml:1273]: Are these relabel configs necessary?

File: cost-analyzer/charts/prometheus/values.yaml
Comment (dwbrown2) [reply cost-analyzer/charts/prometheus/values.yaml:1264]: Are we sure this metric is still alpha? I haven't chased all the way but found this: https://github.com/kubernetes/kubernetes/pull/51553

I think these decisions should be situational. I personally think adding a metric here is valuable as with our past experience it was accurate the vast majority of the time. I do think we should be resilient in the application experience to it being not available on a cluster. We should also consider marking this feature beta in the UI.

File: cost-analyzer/charts/prometheus/values.yaml
Comment (dwbrown2) [inline cost-analyzer/charts/prometheus/values.yaml:1226]: I would have expected this PR to simply add a new metric to the whitelist, but looks like we need to add an entirely new scrape endpoint. Can you share more?

File: cost-analyzer/charts/prometheus/values.yaml
Comment (michaelmdresser) [reply cost-analyzer/charts/prometheus/values.yaml:1264]: I have a K8s v1.22 GKE cluster:
```
→ k version 
...
Server Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.12-gke.2300", GitCommit:"e55564cf3a1384026a54920174977659c8c56a50", GitTreeState:"clean", BuildDate:"2022-08-16T09:24:51Z", GoVersion:"go1.16.15b7", Compiler:"gc", Platform:"linux/amd64"}
...
```

When I look for this metric (its a standard kubelet metric, so at a node's kubelet's /metrics, I see that its an ALPHA status:

```
kubectl proxy --port 8033

curl 'localhost:8033/api/v1/nodes/gke-michael-dev-pool-1-92180698-v5zo/proxy/metrics' | grep -i 'kubelet_volume_stats_used_bytes'

...
# HELP kubelet_volume_stats_used_bytes [ALPHA] Number of used bytes in the volume
# TYPE kubelet_volume_stats_used_bytes gauge
```

Do you know if it has moved out of ALPHA status since 1.22? That PR you linked was merged before v1.22.

File: cost-analyzer/charts/prometheus/values.yaml
Comment (AjayTripathy) [reply cost-analyzer/charts/prometheus/values.yaml:1226]: I tend to agree. @nickcurie can you check if it's on the cadvisor metrics we already scrape? You can learn about that job here: https://github.com/kubecost/cost-analyzer-helm-chart/blob/develop/cost-analyzer/charts/prometheus/values.yaml#L1165

File: cost-analyzer/charts/prometheus/values.yaml
Comment (AjayTripathy) [reply cost-analyzer/charts/prometheus/values.yaml:1264]: I think it's ok to put an alpha metric in but concur that resiliency when the metric is not available is important.

File: cost-analyzer/charts/prometheus/values.yaml
Comment (michaelmdresser) [reply cost-analyzer/charts/prometheus/values.yaml:1226]: This metric isn't a cadvisor metric, its a kubelet metric. See [docs](https://kubernetes.io/docs/concepts/cluster-administration/system-metrics/#metrics-in-kubernetes): "In most cases metrics are available on `/metrics` endpoint of the HTTP server ... Note that kubelet also exposes metrics in `/metrics/cadvisor`, `/metrics/resource` and `/metrics/probes` endpoints. Those metrics do not have same lifecycle."

I'm not sure if we already scrape the standard kubelet metrics or not, just wanted to point this out so we're on the same page. The query I ran in https://github.com/kubecost/cost-analyzer-helm-chart/pull/1713#discussion_r984888548 shows which metrics endpoint this comes from.

Comment (nickcurie) [reply cost-analyzer/charts/prometheus/values.yaml:1226]: Unfortunately, `/metrics/cadvisor` doesn't have any metrics on PVs or PVCs. `/metrics` does, so that's why we need to scrape this endpoint. I'm not sure we have another way to monitor PV usage.

Comment (AjayTripathy) [reply cost-analyzer/charts/prometheus/values.yaml:1226]: SG, just wanted to make sure we checked.

**Review by michaelmdresser — APPROVED** [review]
Looks good, make sure to nuke those unnecessary relabel configs before merging. We have approval to use the alpha metric.

**Comment by michaelmdresser** [comment]
Maybe just add a comment on the prom scrape for this metric that its in alpha.

**Pull request merged**
